concept/type_/type_manager/type_writer.rs

```diff
@@ -30,12 +34,30 @@ pub struct TypeWriter<Snapshot: WritableSnapshot> {
 
 // TODO: Make everything pub(super) and make this submodule of type_manager.
 impl<Snapshot: WritableSnapshot> TypeWriter<Snapshot> {
+    pub(crate) fn storage_put_struct(
+        snapshot: &mut Snapshot,
+        definition_key: DefinitionKey<'static>,
+        struct_definition: StructDefinition,
+    ) {
+        let index_key = LabelToStructDefinitionIndex::build::<BUFFER_KEY_INLINE>(StringBytes::build_ref(
+            struct_definition.name.as_str(),
+        ));
+        snapshot.put_val(
+            index_key.into_storage_key().into_owned_array(),
+            ByteArray::copy(definition_key.clone().into_bytes().bytes()),
```

Comment:
If we're converting an already Cloned structure `into_bytes()`, then why do we need to induce another `ByteArray::copy`? Feels like one copy here is redundant.